lib/create_cg_from_pdb.py

```diff
@@ -1,9 +1,10 @@
-import numpy as np
-import pandas as pd 
+import sys
```

Comment:
@jngrad @paobtorres  I am not completely sure where is the issue but this script has changed its output. The following line
```bash
python3 lib/create_cg_from_pdb.py --download_pdb 1F6S --model 2beadAA --chain A
```
should create the same coarse-grained structure than the one we have stored in `parameters/globular_proteins/1f6s.vtf`

However, now it produces something different.

Original:
```
atom 1 name n resname n resid 1 chain A radius 0.7
atom 2 name E resname GLU resid 1 chain A radius 1.3735
atom 3 name CA resname GLU resid 1 chain A radius 0.7
atom 4 name CA resname GLN resid 2 chain A radius 0.7
atom 5 name Q resname GLN resid 2 chain A radius 1.4504
```
Current:
```
atom 1 name n resname n resid 1 chain A radius 0.7
atom 2 name E resname GLU resid 1 chain A radius 1.3735
atom 3 name CA resname GLU resid 1 chain A radius 0.7
atom 4 name Q resname GLN resid 2 chain A radius 1.4504
atom 5 name CA resname GLN resid 2 chain A radius 0.7
```
Note that the order of atom 5 and 4 have been reversed. The resulting structure looks the same as before by visual inspection in VMD but when I ran  the script in  `samples/Beyer2024/globular_protein.py` with the new structure, it breaks:
```bash
python3 samples/Beyer2024/globular_protein.py --pdb 1f6s --pH 2 --path_to_cg parameters/globular_proteins/1f6s.vtf --mode test
Traceback (most recent call last):
  File "/home/pablo/Desktop/pyMBE/samples/Beyer2024/globular_protein.py", line 164, in <module>
    pmb.create_protein(name=protein_name,
  File "/home/pablo/Desktop/pyMBE/pyMBE.py", line 1053, in create_protein
    particle_id = self.create_particle(name=residue_name,
  File "/home/pablo/Desktop/pyMBE/pyMBE.py", line 955, in create_particle
    raise ValueError(f"{name} must correspond to a label of a pmb_type='particle' defined on df")
ValueError: Ca must correspond to a label of a pmb_type='particle' defined on df
```
`create_cg_from_pdb.py`  also throws the following error, which I am not sure if it was before:

```bash
home/pablo/Desktop/pyMBE/lib/create_cg_from_pdb.py:282: FutureWarning: 
The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. 
In a future version of pandas, the provided callable will be used directly.
 To keep current behavior pass the string "sum" instead.
```

Reply:
I get this script to fail on the main branch too:
```
$ python3 samples/Beyer2024/globular_protein.py --pdb 1f6s --pH 2 --path_to_cg parameters/globular_proteins/1f6s.vtf --mode test
Traceback (most recent call last):
  File "/work/jgrad/pyMBE/samples/Beyer2024/globular_protein.py", line 163, in <module>
    pmb.create_protein(name=protein_name,
  File "/work/jgrad/pyMBE/pyMBE.py", line 1028, in create_protein
    particle_id = self.create_particle(name=residue_name,
  File "/work/jgrad/pyMBE/pyMBE.py", line 930, in create_particle
    raise ValueError(f"{name} must correspond to a label of a pmb_type='particle' defined on df")
ValueError: Ca must correspond to a label of a pmb_type='particle' defined on df
```
Printing the values that already exist in the data frame column reveals the following:
```
['CA' 'n' 'E' 'Q' 'L' 'T' 'K' 'C' 'V' 'F' 'R' 'D' 'Y' 'S' 'P' 'W' 'H' 'A'
 'I' 'N' 'M' 'c' 'AA-n' 'AA-E' 'AA-Q' 'AA-L' 'AA-T' 'AA-K' 'AA-C' 'AA-V'
 'AA-F' 'AA-R' 'AA-D' 'AA-Y' 'AA-S' 'AA-P' 'AA-W' 'AA-H' 'AA-A' 'AA-I'
 'AA-N' 'AA-M' 'AA-c' '1f6s' 'Na' 'Cl']
```
Could this be an issue with case-sensitive names? "CA" may refer to either Ca<sup>2+</sup> or C-alpha carbons, both of which are present in PDB id 1f6s.

Regarding the `FutureWarning`, I have no opinion, since I'm not familiar with this specific part of the pyMBE code. I would be surprised if this was introduced by my changes, since in that function I only adapted the indentation level to 4-characters instead of 8-characters.

Reply:
You are absolutely right @jngrad that was the issue. I checked and for that structure one needs to run the script providing the name of the metal ion 
```bash
python3 samples/Beyer2024/globular_protein.py --pdb 1f6s --pH 2 --path_to_cg parameters/globular_proteins/1f6s.vtf --mode test  --metal_ion_name Ca --metal_ion_charge 2
```
with this line the script runs both in the main branch and this PR.

I am still a bit confused with the change in output of  `lib/create_cg_from_pdb.py`, but it seems harmless as the CI tests passes with the new structure. In the future, when we continue developing the features for globular proteins we need a more robust way of dealing with metal ions and other heterogenous elements in PDB.

Reply:
@pm-blanco @jngrad  I tested both `create_cg_from_pdb.py` and `globular_protein.py` (after creating a new cg model of 1f6s) and both worked okay. 
Regarding the `FutureWarning` I believe its something that started to show up since the later version of pandas.

Reply:
> we need a more robust way of dealing with metal ions and other heterogenous elements in PDB

Looking at [Table 1](https://pubs.acs.org/doi/10.1021/acs.jproteome.0c00906#tbl1) in [10.1021/acs.jproteome.0c00906](https://doi.org/10.1021/acs.jproteome.0c00906), we might as well add the complete periodic table in pyMBE :smile: 

One could also parse the PDBx metadata in CIF files to list metallic ions, and when relevant (e.g. iron) their oxidation state:

* [1F6S.cif](https://files.rcsb.org/header/1F6S.cif)
   ```
   1 polymer     nat ALPHA-LACTALBUMIN 14202.048 6  ? ? ? ? 
   2 non-polymer syn 'CALCIUM ION'     40.078    6  ? ? ? ? 
   3 water       nat water             18.015    89 ? ? ? ? 
   ...
   CA  non-polymer         . 'CALCIUM ION'   ? 'Ca 2'           40.078  
   ```
* [5ZTA.cif](https://files.rcsb.org/header/5ZTA.cif)
   ```
   1 polymer     man 'Sirohydrochlorin ferrochelatase' 30237.533 2  4.99.1.4 ? ? ? 
   2 non-polymer syn 'FE (III) ION'                    55.845    2  ?        ? ? ? 
   3 water       nat water                             18.015    13 ?        ? ? ? 
   ...
   FE  non-polymer         . 'FE (III) ION'  ? 'Fe 3'           55.845  
   ```

Reply:
@jngrad @pm-blanco  I think this is a great idea! I can do it, because at least for proteins we should expand the metal ions supported. We can later discuss more about adding this.